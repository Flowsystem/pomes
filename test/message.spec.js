import React from 'react';
import { mount } from 'enzyme';
import Message from 'message';
import I18nProvider from 'component/component';
import toJson from 'enzyme-to-json';

describe('Message', () => {
  it('render', () => {
    const translations = {};
    const message = mount(
      <I18nProvider translations={translations} lang="en" initialLang="en" initialized>
        <Message id="Hello {name}" values={{ name: 'foo' }} comment="Foo" />
      </I18nProvider>,
    );

    expect(toJson(message)).toMatchSnapshot();
  });
});
