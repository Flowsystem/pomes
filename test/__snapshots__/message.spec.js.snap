// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message render 1`] = `
<I18nProvider
  dispatch={null}
  fallbackLang={null}
  initialLang="en"
  initialized={true}
  lang="en"
  translations={Object {}}
  useReducer={false}
>
  <Message
    comment="Foo"
    id="Hello {name}"
    pluralCondition={null}
    pluralId={null}
    values={
      Object {
        "name": "foo",
      }
    }
  >
    Hello foo
  </Message>
</I18nProvider>
`;
